tests/test_mem_eff_attention.py

```diff
@@ -721,3 +724,85 @@ def test_memory_efficient_attention_full_block_masked(
     assert_allclose(grad_q, query.grad, "grad_q", atol=atol)
     assert_allclose(grad_k, key.grad, "grad_k", atol=atol)
     assert_allclose(grad_v, value.grad, "grad_v", atol=atol)
+
+
+@pytest.mark.parametrize("contiguous", [True, False])
+@pytest.mark.parametrize("dim", [0, 1, 2, 3, 4])
+def test_unbind(dim: int, contiguous: bool):
+    x = contiguous_x = torch.ones([10, 20, 4, 10, 3], requires_grad=True)
+    x2 = contiguous_x2 = torch.ones([10, 20, 4, 10, 3], requires_grad=True)
+
+    if not contiguous:
+        x = x.transpose(-2, -3)
+        x2 = x2.transpose(-2, -3)
```

Comment:
Can I ask for one more test here? (Sorry for being picky).

Would it be possible to do instead something like
```suggestion
    if not contiguous:
        perm = torch.randperm(5).tolist()
        x = x.permute(perm)
        x2 = x2.permute(perm)
```
for more thorough testing of the contiguous / non-contiguous cases ?